### PR TITLE
iMX8MQ SPL/FIT secure boot

### PR DESCRIPTION
**Pull request by igoropaniuk**

### CODE REVIEW

**Review by mike-scott — CHANGES_REQUESTED** [review]

**Comment by mike-scott** [comment]
@igoropaniuk If you would prefer to change all of the UBOOT_MACHINE lines at once in a separate PR, I'm also ok with that.

File: meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc [hunk 1]
```diff
@@ -273,21 +273,29 @@ PREFERRED_PROVIDER_u-boot_mx8mq ?= "u-boot-fio"
...
 PREFERRED_PROVIDER_virtual/kernel_mx8mq ?= "linux-lmp-fslc-imx"
-WKS_FILE_mx8mq_sota = "sdimage-imx8-sota.wks.in"
+WKS_FILE_sota_mx8mq = "sdimage-imx8-spl-sota.wks.in"
+IMXBOOT_TARGETS_mx8mq = "flash_evk_spl"
 ## iMX8MQ EVK
+OSTREE_KERNEL_ARGS_imx8mqevk ?= "console=tty1 console=ttymxc0,115200 earlycon=ec_imx6q,0x30860000,115200 root=/dev/mmcblk0p2 rootfstype=ext4"
 SOTA_CLIENT_FEATURES_append_imx8mqevk = " ubootenv"
 MACHINE_FEATURES_remove_imx8mqevk = "qca6174"
 MACHINE_FEATURES_append_imx8mqevk = " bcm4356"
+UBOOT_MACHINE_imx8mqevk = "imx8mq_evk_config"
```
[the remark this answers is not in the document]
Comment (igoropaniuk) [reply meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc:298]: Fixed and re-pushed

**Comment by igoropaniuk** [comment]
@mike-scott I've fixed that line for imx8mq, others we can address in a separate PR

File: meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc (hunk 1 above)
[the remark this answers is not in the document]
Comment (mike-scott) [reply meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc:298]: I don't see this as changed on my end?

**Comment by mike-scott** [comment]
@igoropaniuk doesn't look like this was re-pushed?

File: meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc (hunk 1 above)
[the remark this answers is not in the document]
Comment (igoropaniuk) [reply meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc:298]: hm, really strange
```
lmp.repo/layers/meta-lmp$ git push -f igoropaniuk imx8mq_secure_boot 
Everything up-to-date

lmp.repo/layers/meta-lmp$ git show HEAD~1 | grep UBOOT_MACHINE_imx8mqevk
+UBOOT_MACHINE_imx8mqevk ?= "imx8mq_evk_config"

lmp.repo/layers/meta-lmp$ git remote -v
fio	https://github.com/foundriesio/meta-lmp (fetch)
fio	https://github.com/foundriesio/meta-lmp (push)
igoropaniuk	git@github.com:igoropaniuk/meta-lmp.git (fetch)
igoropaniuk	git@github.com:igoropaniuk/meta-lmp.git (push)

```

**Review by mike-scott — APPROVED** [review]
LGTM

**Review by ricardosalveti — APPROVED** [review]
LGTM

File: meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc [hunk 2]
```diff
@@ -273,21 +273,29 @@ PREFERRED_PROVIDER_u-boot_mx8mq ?= "u-boot-fio"
 PREFERRED_PROVIDER_u-boot-fw-utils_mx8mq = "libubootenv"
 PREFERRED_RPROVIDER_u-boot-fw-utils_mx8mq = "libubootenv"
 UBOOT_DTB_LOADADDRESS_mx8mq = "0x43000000"
-IMAGE_BOOT_FILES_mx8mq = "imx-boot ${KERNEL_IMAGETYPE} ${@make_dtb_boot_files(d)} boot.scr uEnv.txt"
-IMAGE_BOOT_FILES_sota_mx8mq = "imx-boot boot.itb"
+PREFERRED_PROVIDER_virtual/trusted-firmware-a_mx8mm ?= "imx-atf"
```
Comment (mike-scott) [inline meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc:276]: Shouldn't this be for mx8mq not mx8mm?
```
PREFERRED_PROVIDER_virtual/trusted-firmware-a_mx8mq ?= "imx-atf"
```

**Review by mike-scott — CHANGES_REQUESTED** [review]

File: meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc (hunk 2 above)
Comment (mike-scott) [reply meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc:276]: Also title has a typo:
`bsp: lmp-machine-custom: mx8qm: changes for imx-boot and uboot-fitmage class`
should be:
`bsp: lmp-machine-custom: mx8mq: changes for imx-boot and uboot-fitmage class`
(change mx8qm to mx8mq)

Comment (igoropaniuk) [reply meta-lmp-bsp/conf/machine/include/lmp-machine-custom.inc:276]: nice catch, fixed!

**Review by mike-scott — APPROVED** [review]

**Pull request merged**